app/main/lib/shared_models/video_model.py

```diff
@@ -14,57 +15,100 @@
-                        existing.context.append(video.context)
-                    flag_modified(existing, 'context')
-                saved_video = existing
-        except NoResultFound as e:
-            # Otherwise, add new video, but with context as an array
-            if video.context and not isinstance(video.context, list):
-                video.context = [video.context]
-            db.session.add(video)
-            saved_video = video
-        except Exception as e:
-            db.session.rollback()
-            raise e
-        try:
-            db.session.commit()
-            return saved_video
-        except Exception as e:
-            db.session.rollback()
-            raise e
+    def overload_context_to_denote_content_type(self, task):
+        return {**task, **{"context": {**task.get("context", {}), **{"content_type": "video"}}}}
+
+    def delete(self, task):
+        return media_crud.delete(task, Video)
+
+    def add(self, task):
+        s3_folder = task["hash_value"]["folder"]
+        s3_filepath = task["hash_value"]["filepath"]
+        task["hash_value"] = task["hash_value"]["hash_value"]
+        app.logger.error(f"Task looks like: {task}")
+        added, obj = media_crud.add(task, Video, ["hash_value"])
+        download_file_from_s3(s3_folder, s3_filepath, media_crud.tmk_file_path(obj.folder, obj.filepath))
+        return added
+
+    def store_audio(self, task):
+        if task.get("match_across_content_types", True):
+            am = AudioModel('audio')
+            am.add(self.overload_context_to_denote_content_type(task))
+            audio_matches = am.blocking_search(task, "audio")
+            return audio_matches["result"]
+        return []
+
+    def blocking_search(self, task, modality):
+        video, temporary, context, presto_result = media_crud.get_blocked_presto_response(task, Video, modality)
+        video.hash_value = presto_result["body"]["hash_value"]
+        if video:
+            matches = self.search(task, context[0], True).get("result")
+            for match in self.store_audio(task):
+                matches.append(match)
+            if temporary:
+                media_crud.delete(task, Video)
+            else:
+                media_crud.save(video, Video, ["hash_value"])
+            matches.sort(key=lambda x: x['score'], reverse=True)
+            if task.get("limit"):
+                return {"result": matches[:task.get("limit")]}
+            else:
+                return {"result": matches}
+        else:
+            return {"error": "Video not found for provided task", "task": task}
+
+    def async_search(self, task, modality):
+        return media_crud.get_async_presto_response(task, Video, modality)
 
     def get_tempfile(self):
         return tempfile.NamedTemporaryFile()
 
-    def execute_command(self, command):
-        return os.popen(command).read()
+    def execute_command(self, cmd, params):
+      return db.session.execute(cmd, params).fetchall()
```

Comment:
We have been doing this for years at this point. I don't know if this is the time to solve this. The reason this is so complicated is because of `get_context_query(context, value_as_json=False, vars_as_hash=True)` which is a complicated mess that does embedded JSON context querying. It is complicated. I agree that its suboptimal but I don't know if this is the time to solve this problem.